### PR TITLE
Implementation of S3M's Handlers Analysis

### PR DESCRIPTION
**Pull request by jvcoutinho**
Implements Mining Framework's base interfaces for analysis of multiple handlers of [S3M](https://github.com/guilhermejccavalcanti/jFSTMerge), with an initial focus for the renaming handlers.

Notes:

- I've extended TravisHelper to be able to give me the token after its creation, but I **have not** included this change in this PR, instead I've opened #76.
- Please, don't integrate the PR until the study is finished.

### CODE REVIEW

File: src/services/S3MHandlersAnalysis/datacollection/S3MRunner.groovy [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            runS3M(leftFile, baseFile, rightFile, 'SF.java', Handlers.Renaming, '-r', 'SAFE')
+            runS3M(leftFile, baseFile, rightFile, 'MM.java', Handlers.Renaming, '-r', 'MERGE')
+            runS3M(leftFile, baseFile, rightFile, 'KB.java', Handlers.Renaming, '-r', 'BOTH')
+        }
+    }
+
+    private static void runS3M(Path leftFile, Path baseFile, Path rightFile, String outputFileName, Handlers handler, String... additionalParameters) {
+        Process S3M = ProcessRunner.startProcess(buildS3MProcess(leftFile, baseFile, rightFile, outputFileName, handler, additionalParameters))
+        S3M.getInputStream().eachLine {
+            //println it
```
Comment (rafaelmotaalves) [inline src/services/S3MHandlersAnalysis/datacollection/S3MRunner.groovy]: commented out code, should this loop be removed?

File: src/services/S3MHandlersAnalysis/implementations/ProjectProcessor.groovy [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package services.S3MHandlersAnalysis.implementations
+
+import main.exception.TravisHelperException
+import main.project.Project
+import main.util.GithubHelper
+import main.util.TravisHelper
+
+import static main.app.MiningFramework.arguments
+
+class ProjectProcessor implements main.interfaces.ProjectProcessor {
```
Comment (rafaelmotaalves) [inline src/services/S3MHandlersAnalysis/implementations/ProjectProcessor.groovy]: this class is the same as the other research

File: src/services/S3MHandlersAnalysis/datacollection/S3MRunner.groovy (hunk 1 above)
Comment (jvcoutinho) [reply src/services/S3MHandlersAnalysis/datacollection/S3MRunner.groovy]: Without "draining" all of the output, the process never finishes.

Comment (rafaelmotaalves) [reply src/services/S3MHandlersAnalysis/datacollection/S3MRunner.groovy]: running S3M.waitFor() woudn't wait for it to finish?

Comment (jvcoutinho) [reply src/services/S3MHandlersAnalysis/datacollection/S3MRunner.groovy]: Yes, but if the process has bufferized output, it won't finish. Here's a [link](https://stackoverflow.com/questions/5483830/process-waitfor-never-returns) about this.

File: src/services/S3MHandlersAnalysis/implementations/ProjectProcessor.groovy [hunk 3]
```diff
@@ -0,0 +1,72 @@
+package services.S3MHandlersAnalysis.implementations
```
Comment (rafaelmotaalves) [inline src/services/S3MHandlersAnalysis/implementations/ProjectProcessor.groovy:1]: Essa classe parece fazer os mesmo da ForkAndEnableTravisProcessor, se realmente for, usar a ForkAndEnableTravisProcessor invés dela no injector

**Comment by rafaelmotaalves** [comment]
Após resolver os comentários, integrar a master nessa branch e corrigir a estrutura de pastas para a atual do framework, em que as classes estão divididas por tipo e não por injector

**Pull request merged**